### PR TITLE
NIAD-826: CI PR build pipeline

### PR DESCRIPTION
**Pull request by antonlebedjko**
This is the initial Jenkins set up. However, some updates will be required once the AWS is ready. Right now we are just printing the errors because the AWS is not ready yet but we want to throw the errors once it is. If AWS becomes available before this PR is merged, we will update this PR. Otherwise the new ticket will be needed.

Co-authored with @andrewwa-kainos

### CODE REVIEW

**Review by saileshp-kainos — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+// Retrieving Secrets from AWS Secrets
+String getSecretValue(String secretName, String region) {
+  String awsCommand = "aws secretsmanager get-secret-value --region ${region} --secret-id ${secretName} --query SecretString --output text"
+  return sh(script: awsCommand, returnStdout: true).trim()
+}
+
+Map<String,Object> decodeSecretKeyValue(String rawSecret) {
+  List<String> secretsSplitted = rawSecret.replace("{","").replace("}","").split(",")
+  Map<String,Object> secretsDecoded = [:]
+  secretsSplitted.each {
```
Comment (n-r-b) [inline Jenkinsfile:180]:
```suggestion
  List<String> secretsSplit = rawSecret.replace("{","").replace("}","").split(",")
  Map<String,Object> secretsDecoded = [:]
  secretsSplit.each {
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+int terraform(String action, String tfStateBucket, String project, String environment, String component, String region, Map<String, String> variables=[:], List<String> parameters=[]) {
+    println("Running Terraform ${action} in region ${region} with: \n Project: ${project} \n Environment: ${environment} \n Component: ${component}")
+    variablesMap = variables
+    variablesMap.put('region',region)
+    variablesMap.put('project', project)
+    variablesMap.put('environment', environment)
+    variablesMap.put('tf_state_bucket',tfStateBucket)
+    parametersList = parameters
+    parametersList.add("-no-color")
+    //parametersList.add("-compact-warnings")  /TODO update terraform to have this working
```
Comment (n-r-b) [inline Jenkinsfile:146]: Is this something we plan to do? I noticed it's also commented out in the NHAIS project

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+              lock("${tfProject}-${tfEnvironment}-${tfComponent}")
+            }
+            stages {
+                stage('Deploy using Terraform') {
+                    steps {
+                        script {
+                            String tfCodeBranch  = "develop"
+                            String tfCodeRepo    = "https://github.com/nhsconnect/integration-adaptors"
+                            String tfRegion      = "${TF_STATE_BUCKET_REGION}"
+                            List<String> tfParams = []
```
Comment (n-r-b) [inline Jenkinsfile:94]: I don't think this is used anywhere?

File: Jenkinsfile (hunk 2 above)
Comment (antonlebedjko) [reply Jenkinsfile:146]: @bartek-sarul do we have any plans about updating the terraform? Or it's ok to just remove it?

File: Jenkinsfile (hunk 3 above)
Comment (antonlebedjko) [reply Jenkinsfile]: @n-r-b well spotted! Has been removed now

File: Jenkinsfile (hunk 2 above)
Comment (andrewwa-kainos) [reply Jenkinsfile:145]: As per @michael-kainos's advice, we can remove this

Comment (antonlebedjko) [reply Jenkinsfile]: Has been fixed in c00dbc4b58360efd8151c61f292e624c1f44fcd4

**Review by n-r-b — DISMISSED** [review]
Looks good!

**Review by andrewwa-kainos — APPROVED** [review]

**Review by n-r-b — APPROVED** [review]

**Pull request merged**